docs: update synthetics with zip support (#609)
elastic/observability-docs: docs/en/observability/synthetics-quickstart.asciidoc

There are two ways to configure a synthetic test. The config file below demonstrates both.
The first entry in `heartbeat.monitors` uses the `inline` source type, where the entire synthetic journey is directly embedded in the yaml file.

> The second method, uses the `zip_url` source type, which points to remote endpoint where synthetics tests can be downloaded with basic authentication.

Reviewer: ```suggestion
The second method uses the `zip_url` source type, which points to a remote endpoint where synthetics tests can be downloaded with basic authentication.
```

  type: browser
  schedule: "@every 1m"
  source:
    zip_url: <3>
      url: "https://github.com/elastic/synthetics/archive/refs/heads/master.zip" <4>
      folder: "examples/todos" <5>
      username: "" <6>
      password: "" 
----
<1> Each `monitor` gets its own ID in the {uptime-app} and, therefore its own schedule entry.
This allows tests to be run in parallel and analyzed separately.
<2> In this example, a synthetic test is defined inline. This is a two-step script that first loads
a homepage and then hovers over a product menu. See <<synthetics-syntax>> for more information.

> <3> In this example, our library of synthetic tests is downloaded from the

Reviewer 2: Super nit and subjective:

```suggestion
<3> In this example, we download our library of synthetic tests from the
```

The tech writers have usually recommended that I avoid passive language, so this simplification is worth considering. We could also write `"This example downloads..."`

Reviewer 3: That's a great point Justin however I think this is a different situation because we don't download the library it is downloaded by heartbeat at some later point on a schedule. At least that's how it reads to me. @dedemorton WDYT?

remote zip endpoint for https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. 

> <4> Note that the url refers to the endpoint where the test project exists.

Reviewer 2: Could we highlight these config keywords as code? Showing `url`, `folder`, `username`, `password` as code will add a subliminal element to this section that would make it easier for me to understand.

Then we could go back to lowercase "folder" and change "username" to be lower as well in item 6.

Author: Thanks Justin, Missed your review, will address it with a new PR.

  type: browser
  schedule: "@every 1m"
  source:
    zip_url: <3>
      url: "https://github.com/elastic/synthetics/archive/refs/heads/master.zip" <4>
      folder: "examples/todos" <5>
      username: "" <6>
      password: "" 
----
<1> Each `monitor` gets its own ID in the {uptime-app} and, therefore its own schedule entry.
This allows tests to be run in parallel and analyzed separately.
<2> In this example, a synthetic test is defined inline. This is a two-step script that first loads
a homepage and then hovers over a product menu. See <<synthetics-syntax>> for more information.
<3> In this example, our library of synthetic tests is downloaded from the

> remote zip endpoint which is https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. 

Reviewer: ```suggestion
remote zip endpoint, which is https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. 
```

> remote zip endpoint which is https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. 

Reviewer 3: ```suggestion
remote zip endpoint for https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. 
```

> <4> Note that the url refers to the endpoint where the test projects exists.

Reviewer: should say either "test project exists" or "test projects exist" (depends on whether there is one or many projects)

Author: Thanks, fixed now.

> <4> Note that the url refers to the endpoint where the test projects exists.

Author: ```suggestion
<4> Note that the url refers to the endpoint where the test project exists.
```

> <5> folder refers to the relative path where the synthetic journey files are

Reviewer: ```suggestion
<5> `folder` refers to the relative path where the synthetic journey files are
```

> <5> folder refers to the relative path where the synthetic journey files are

Reviewer 3: ```suggestion
<5> Folder refers to the relative path where the synthetic journey files are located
```

located. {heartbeat} will invoke the synthetics library on this folder.

> <6> Username and password if provided will be sent as HTTP Basic Authentication headers to the remote zip endpoint.

Reviewer: ```suggestion
<6> Username and password, if provided, will be sent as HTTP Basic Authentication headers to the remote zip endpoint.
```

> <6> Username and password if provided will be sent as HTTP Basic Authentication headers to the remote zip endpoint.

Reviewer 3: ```suggestion
<6> Username and password are blank here, but if provided will be sent as HTTP Basic Authentication headers to the remote zip endpoint.
```
